context-linux/src/etc/one-context.d/net-96-cloudinit-finish

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+# -------------------------------------------------------------------------- #
+# Copyright 2002-2024, OpenNebula Project, OpenNebula Systems                #
+#                                                                            #
+# Licensed under the Apache License, Version 2.0 (the "License"); you may    #
+# not use this file except in compliance with the License. You may obtain    #
+# a copy of the License at                                                   #
+#                                                                            #
+# http://www.apache.org/licenses/LICENSE-2.0                                 #
+#                                                                            #
+# Unless required by applicable law or agreed to in writing, software        #
+# distributed under the License is distributed on an "AS IS" BASIS,          #
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   #
+# See the License for the specific language governing permissions and        #
+# limitations under the License.                                             #
+#--------------------------------------------------------------------------- #
+
```

Comment:
```ruby
if [ -z "$USER_DATA" ]; then
    exit 0
fi
```
Something like this will prevent execution when no userdata is provided, I think the sooner we exit the better. :thinking: In this case it prevents uncoditional lock-up when multistep build like done in https://github.com/OpenNebula/marketplace-community/tree/master/apps-code. This "lock-up" happens below at `trap lock_and_cleanup EXIT`m which can be considered a bug.

Reply:
Yep, I didn't take into account multi-stage builds, so early returning when is no USER_DATA will make more sense 👍🏼

Reply:
Also added a test in https://github.com/OpenNebula/development/pull/2639/files#diff-6c36a1536fc2200ff672a04d56670696d6b895cf1a95b33b43c2b1961a1e5451R91-R96